src/Builder.jl

```diff
@@ -104,36 +97,23 @@ function Selectors.runner(::Type{SetupBuildDirectory}, doc::Documents.Document)
             end
             for file in files
                 src = normpath(joinpath(root, file))
-                dst = normpath(joinpath(build, relpath(root, source), file))
+                path = normpath(joinpath(relpath(root, source), file)) # path relative to source
+                dst = normpath(joinpath(build, path))
                 # Non-markdown files are simply copied over to `build`.
                 # Markdown files get added to the document tree as `Page` objects.
-                endswith(src, ".md") ?
-                    Documents.addpage!(doc, src, dst) :
+                if endswith(file, ".md")
+                    name = first(splitext(path))
+                    Documents.addpage!(doc, name, src, dst)
+                else
                     cp(src, dst; remove_destination = true)
+                end
```

Comment:
I think I still disagree, but, after sleeping on it, I do see that `addpage!` can have alternative semantics, where this makes sense (and my original change seems to have been changing the semantics). Also, links are an issue I didn't really consider, so I need to put in a bit more thought (i.e. beyond this PR).

Changed to the code you proposed.